great_expectations/rule_based_profiler/domain_builder/categorical_column_domain_builder.py

```diff
@@ -0,0 +1,247 @@
+        # rest of the metrics.
+        table_column_names: List[str] = self._get_table_column_names_from_first_batch(
+            validator=validator,
+            batch_ids=batch_ids,
+        )
+
+        metrics_for_cardinality_check: List[
+            Dict[str, List[MetricConfiguration]]
+        ] = self._generate_metric_configurations_to_check_cardinality(
+            batch_ids=batch_ids, column_names=table_column_names
+        )
+
+        candidate_column_names: List[str] = self._columns_meeting_cardinality_limit(
+            validator=validator,
+            table_column_names=table_column_names,
+            metrics_for_cardinality_check=metrics_for_cardinality_check,
+        )
+
+        column_domains: List[Domain] = build_domains_from_column_names(
+            candidate_column_names
+        )
+
+        return column_domains
+
+    def _get_table_column_names_from_first_batch(
+        self,
+        validator: Validator,
+        batch_ids: List[str],
+    ):
+        """Retrieve table column names from the first loaded batch.
+
+        Args:
+            validator: Validator to use in retrieving columns.
+            batch_ids: Batches used in this profiler rule.
+
+        Returns:
+            List of column names from the first loaded batch.
+        """
+
+        if len(batch_ids) == 0:
+            raise ProfilerConfigurationError(
+                "No batch_ids were available, please check your configuration."
+            )
+
+        first_batch_id: str = batch_ids[0]
+
+        table_column_names: List[str] = validator.get_metric(
+            metric=MetricConfiguration(
+                metric_name="table.columns",
+                metric_domain_kwargs={
+                    "batch_id": first_batch_id,
+                },
+                metric_value_kwargs=None,
+                metric_dependencies=None,
+            )
+        )
+
+        if self._exclude_columns is not None:
+            table_column_names = [
+                colname
+                for colname in table_column_names
+                if colname not in self._exclude_columns
+            ]
+
+        return table_column_names
+
+    def _generate_metric_configurations_to_check_cardinality(
+        self,
+        batch_ids: List[str],
+        column_names: List[str],
+    ) -> List[Dict[str, List[MetricConfiguration]]]:
+        """Generate metric configurations used to compute metrics for checking cardinality.
+
+        Args:
+            batch_ids: List of batch_ids used to create metric configurations.
+            column_names: List of column names used to create metric configurations.
+
+        Returns:
+            List of dicts of the form [{column_name: List[MetricConfiguration]},...]
+        """
+
+        limit_mode: CardinalityLimitMode = self._cardinality_checker.limit_mode
+
+        metric_name: str = limit_mode.related_metric_name
+
+        metric_configurations: List[Dict[str, List[MetricConfiguration]]] = []
```

Comment:
@anthonyburdi The `metric_configurations` can be more elegantly gathered as a comprehension.  Thanks!

Reply:
Good call. I didn't do it initially because it's a large nested block of code but you are right. Changed!